sci-libs/vtk/vtk-9.0.1.ebuild

```diff
@@ -294,7 +294,25 @@ src_configure() {
 	fi
 
 	if use cuda; then
-		mycmakeargs+=( -DVTKm_CUDA_Architecture="native" )
+		local MY_CUDA_ARCH=native
+		if [[ -n ${VTK_CUDA_ARCH} ]]; then
+			# we ignore fermi arch, because current nvidia-cuda-toolkit-11*
+			# no longer supports it
+			case ${VTK_CUDA_ARCH} in
+				kepler|maxwell|pascal|volta|turing|ampere|all|none)
+					MY_CUDA_ARCH=${VTK_CUDA_ARCH}
+					;;
+				*)
+					# use default value of MY_CUDA_ARCH, ignore any other values
```

Comment:
Could `die` with arch's value in the message to inform users they set an unexpected value rather than fallback to native against the user's wish.

May allow to simplify too given this case statement would be a pure sanity check and not need to set anything.

It's unlikely so even falling late doesn't seem so bad.

Reply:
agreed